src/ImageSharp/Memory/Allocators/MemoryAllocator.cs

```diff
@@ -89,13 +101,42 @@ public virtual void ReleaseRetainedResources()
         where T : struct
         => MemoryGroup<T>.Allocate(this, totalLength, bufferAlignment, options);
 
-    internal static void MemoryGuardMustBeBetweenOrEqualTo(int value, int min, int max, string paramName)
+    internal static void MemoryGuardMustBeBetweenOrEqualTo<T>(int value, int min, int max, string paramName)
```

Comment:
`min` is `0` at each call of this method. I think it would be better to redefine this method as a buffer length validator, and throw a different exception (or use a different exception message) for `length < 0`. Also, `length == 0` calls are suspicious but maybe there are valid cases (when not, we should disallow them at higher levels).

Reply:
We allow 0 for some reason, returning an empty buffer. Don't ask me why 🤷